Update ActivityCard to latest hifi designs

ryelle opened the pull request:
This PR updates the ActivityCard to the latest design. Fixes #95  

Mockup:
<img width="519" alt="screen shot 2018-07-13 at 12 47 30 pm" src="https://user-images.githubusercontent.com/541093/42703588-f9d79614-869a-11e8-87cc-ae8fbef46fbb.png">

This PR's implementation:
<img width="514" alt="screen shot 2018-07-13 at 12 49 47 pm" src="https://user-images.githubusercontent.com/541093/42703691-42674fa0-869b-11e8-9989-451cc246adcf.png">

Note: I think the antialiasing on the text is a little different from mockup to browser, so it looks darker here. The timestamp is also intentionally darker, because that does not have enough contrast in the mockup.

Obviously the flag and order status components are not done yet, but the basics are in place now for use in all the activity panels. This also updates [the component readme](https://github.com/woocommerce/wc-admin/blob/115ad9e6079781d199776111b5969073f78f7891/client/layout/activity-panel/activity-card/README.md) with the new props, and adds snapshot tests for the card component.

To test:

- Verify that the orders list renders as expected
- Run the tests locally with `npm test client/layout/activity-panel/activity-card/test/index.js`

ryelle committed:
ActivityCard: Update to match new activity panel designs

ryelle committed:
Add spacing rhythm system, add the rest of the base colors

ryelle committed:
Add styles for the ActivityCard

ryelle committed:
Use the new ActivityCard in the orders panel

ryelle committed:
Update text color & size

ryelle committed:
Disable timestamped test (string changes based on date)

ryelle commented on client/layout/activity-panel/activity-card/test/index.js:
```diff
@@ -0,0 +1,100 @@
+		expect( card ).toMatchSnapshot();
+	} );
+
+	xtest( 'should render a timestamp on a card', () => {
```
I'd like to see if we could leave this test in, but the snapshot stays fixed ("3 days ago") while the actual output of the component changes based on date. Could we mock the date formatter? Would probably want to address in a follow-up PR

timmyc replied:
Could we create a timestamp on the fly in the test here using moment so the date is 3 days ago from today?

timmyc commented on client/layout/activity-panel/activity-card/index.js:
```diff
@@ -4,43 +4,40 @@
 class ActivityCard extends Component {
 	render() {
-		const { actions, className, date, icon, image, label, menu, children } = this.props;
+		const { actions, className, children, date, icon, subtitle, title, unread } = this.props;
```
Thanks for alphabetizing here... 🗑

timmyc commented on client/layout/activity-panel/activity-card/test/__snapshots__/index.js.snap:
```diff
@@ -0,0 +1,248 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
TIL Jest Snapshots. Awesome.

timmyc commented on client/stylesheets/_variables.scss:
```diff
@@ -0,0 +1,11 @@
+
+$gutter: var(--main-gap);
+
+$gap-large: 24px;
```
EXCELLENT!!!

timmyc reviewed (approved):
So. Much. Good. Stuff! Only one suggestion on maybe getting the test to work on the "3 Days Ago" but otherwise this is all fantastic. Testing well for me - travis will become happy if you rebase too. Pre-approvin'

ryelle committed:
Use moment to generate a relative timestamp to match “3 days ago” snapshot

ryelle committed:
Rename spacing vars

The difference between gap and gap-med is not intuitive, so let’s switch to -small, and change -small to -smallest.

timmyc commented:
> The Jest snapshot functionality is really great, we should try using it in more of our components - so much less finicky than trying to test into rendered components for certain classes.

💯 !!

And glad the moment trick worked too!
